### PR TITLE
fix(ci): improve CircleCI job duration

### PR DESCRIPTION
**Pull request by brettlangdon**
This fix introduces a few changes to improve overall CircleCI job runtime.

**Before:**
- Runtime: 28+ minutes
- Credits: 33k

**After:**
- Runtime: ~22 minutes (-20%)
- Credits: ~15k (-55%)

### Fix mysql test pattern

The `mysql` CircleCI job was using the pattern "mysql", which would also match the `mysqldb` jobs.

This change updates the test pattern for all mysql jobs to include an ending `$`.

<details>
<summary>Before</summary>

```
❯ riot list mysql
 No.    Hash     Name     Interpreter                Environment                                               Packages
 #554   1a1c1fa  mysql    Interpreter(_hint='2.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #555   8ba1245  mysql    Interpreter(_hint='2.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.23'
 #556   1e72d9c  mysql    Interpreter(_hint='3.5')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #557   16025dc  mysql    Interpreter(_hint='3.5')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.23'
 #558   1cdb1c5  mysql    Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #559   1230ff8  mysql    Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.29'
 #560   729b5a8  mysql    Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #561   1133f63  mysql    Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python'
 #562   129b0f2  mysql    Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #563   12db871  mysql    Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python'
 #564   270f95a  mysql    Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #565   9182ae0  mysql    Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python'
 #566   137147e  mysql    Interpreter(_hint='3.10')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python~=8.0.28'
 #567   f26af3b  mysql    Interpreter(_hint='3.10')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python'
 #568   2eb00ea  mysql    Interpreter(_hint='3.11')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python~=8.0.31'
 #569   10d623a  mysql    Interpreter(_hint='3.11')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysql-connector-python'
 #1450  9be82d5  mysqldb  Interpreter(_hint='2.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=1.4.6'
 #1451  17a3e91  mysqldb  Interpreter(_hint='3.5')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=1.4.6'
 #1452  1c338bf  mysqldb  Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.0'
 #1453  162a527  mysqldb  Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1454  19ef42f  mysqldb  Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
 #1455  145f708  mysqldb  Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.0'
 #1456  73a2d0b  mysqldb  Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1457  11b2766  mysqldb  Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
 #1458  1fcb3db  mysqldb  Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.0'
 #1459  1c5e6ed  mysqldb  Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1460  137f08a  mysqldb  Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
 #1461  149f779  mysqldb  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.0'
 #1462  14a543a  mysqldb  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1463  1eea232  mysqldb  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
 #1464  14a543a  mysqldb  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1465  1eea232  mysqldb  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
 #1466  10bc6fe  mysqldb  Interpreter(_hint='3.10')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1467  bf00ea5  mysqldb  Interpreter(_hint='3.10')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
 #1468  186525e  mysqldb  Interpreter(_hint='3.11')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient~=2.1'
 #1469  8e13da5  mysqldb  Interpreter(_hint='3.11')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                               'hypothesis<6.45.1' 'mysqlclient'
```

</details>

<details>
<summary>After</summary>

```
❯ riot list "mysql$"
 No.   Hash     Name   Interpreter                Environment                                               Packages
 #554  1a1c1fa  mysql  Interpreter(_hint='2.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #555  8ba1245  mysql  Interpreter(_hint='2.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.23'
 #556  1e72d9c  mysql  Interpreter(_hint='3.5')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #557  16025dc  mysql  Interpreter(_hint='3.5')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.23'
 #558  1cdb1c5  mysql  Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #559  1230ff8  mysql  Interpreter(_hint='3.6')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.29'
 #560  729b5a8  mysql  Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #561  1133f63  mysql  Interpreter(_hint='3.7')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python'
 #562  129b0f2  mysql  Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #563  12db871  mysql  Interpreter(_hint='3.8')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python'
 #564  270f95a  mysql  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python==8.0.5'
 #565  9182ae0  mysql  Interpreter(_hint='3.9')   DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python'
 #566  137147e  mysql  Interpreter(_hint='3.10')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python~=8.0.28'
 #567  f26af3b  mysql  Interpreter(_hint='3.10')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python'
 #568  2eb00ea  mysql  Interpreter(_hint='3.11')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python~=8.0.31'
 #569  10d623a  mysql  Interpreter(_hint='3.11')  DD_TESTING_RAISE=1 DD_REMOTE_CONFIGURATION_ENABLED=false  'mock' 'pytest' 'pytest-mock' 'coverage' 'pytest-cov' 'opentracing'
                                                                                                            'hypothesis<6.45.1' 'mysql-connector-python'
```

</details>

### Improve wait job runtime

The `wait` job was not running with `-s` and was trying to install the dev package this meant it was taking 3+ minutes to run the `wait` job. We updated the job to ensure it does not install the dev package. Now it runs in <30 seconds.

### Properly restore workspace from build_base_venvs

Through debugging we noticed that we were building the base venv/installing dev package in scenarios when we should not have.

This is because we were storing the workspace/venvs from `build_base_venvs` incorrectly causing us to reinstall the dev package once per-job * per-node * per-python version.

The fix was to checkout the code first and then restore the files from the workspace.

The only lead I have as to why/when this started happening is when we added `.riot/requirements/` if the git checkout was overwriting the exiting `.riot/` directory added from the workspace?

### Adjusted parallelism

Now that runtime of jobs is improved we can safe some credits by reducing parallelism.

We combined the `psycopg` and `psycopg2` jobs.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: riotfile.py [hunk 1]
```diff
@@ -577,6 +577,8 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
             command="python tests/wait-for-services.py {cmdargs}",
             # Default Python 3 (3.10) collections package breaks with kombu/vertica, so specify Python 3.9 instead.
             pys="3.9",
+            create=True,
+            skip_dev_install=True,
```
Comment (P403n1x87) [inline riotfile.py:581]: Is this change still needed after the CircleCI changes? I'd expect `-s` to do the job here if we have valid base venvs

Comment (brettlangdon) [reply riotfile.py:581]: I seriously had mixed results with `create` and `skip_dev_install`... better safe than sorry?

Comment (brettlangdon) [reply riotfile.py:581]: but that was also before I realized we weren't restoring the original base venvs properly.

I think I noticed that with `create` if the base venv didn't exist, it would still try to create it and install the dev package even if both of these were set.

**Review by P403n1x87 — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]
